src/main/java/uk/gov/di/ipv/cri/common/library/domain/personidentity/PersonIdentityDetailed.java

```diff
@@ -22,6 +23,9 @@ public class PersonIdentityDetailed {
     @JsonProperty("passport")
     private final List<Passport> passports;
 
+    @JsonProperty("device_information")
```

Comment:
Since this class isn't marked with `@JsonIgnoreProperties(ignoreUnknown = true)` does the caller of this method (assuming external request) send the `device_information`? If not, this will throw so have those places been updated to send device information?

Reply:
Hmm I'm a little confused actually, so looking at [the docs ](https://fasterxml.github.io/jackson-annotations/javadoc/2.7/com/fasterxml/jackson/annotation/JsonInclude.Include.html#NON_NULL) it indicates that null values are ignored, rather than causing errors? If the `device_information` is not set, I want it to be ignored completely

Reply:
`@JsonInclude` controls how nulls are handles. 

The way it's used in your code is if the value is null, it'll be excluded. While `@JsonIgnoreProperties` will not attempt to deserialize a field that isn't present. 

So the difference is, `@JsonInclude` still expects the key to be present but will skip it, if the value is null.

Reply:
Right ok, so are you saying that I need to initialise `device_information` as null to then rely on the setter?

Reply:
Better just to add the annotation, I think that'll be the easiest solution, what do you think?

Reply:
Maybe we do some experiments on a call to see the exact behaviour and then we can cater towards it. So some TDD!